src/components/starter/theme_switcher.tsx

```diff
@@ -0,0 +1,39 @@
+import { FunctionComponent } from 'react';
+import { css } from '@emotion/react';
+import { EuiButtonIcon, EuiToolTip, useEuiTheme } from '@elastic/eui';
+import { useTheme } from '../theme';
+
+/**
+ * Current theme is set in localStorage
+ * so that it persists between visits.
+ */
+const ThemeSwitcher: FunctionComponent = () => {
+  const { colorMode, setColorMode } = useTheme();
+  const isDarkTheme = colorMode === 'dark';
+
+  const handleChangeTheme = (newTheme: string) => {
+    setColorMode(newTheme);
+  };
+
+  const lightOrDark = isDarkTheme ? 'light' : 'dark';
+  const { euiTheme } = useEuiTheme();
+  const IconRotator = isDarkTheme ? '-270deg' : '0deg';
+
+  return (
+    <EuiToolTip content={`Change theme to ${lightOrDark}`} key="theme-switch">
+      <EuiButtonIcon
+        css={css`
+          rotate: ${IconRotator};
```

Comment:
@boriskirov this is not working when is deployed: [https://next-eui-starter-git-elastic-products-miukimiu.vercel.app/](https://next-eui-starter-git-elastic-products-miukimiu.vercel.app/).

Shoud it be instead?
```suggestion
          transform: rotate(${IconRotator});
```
 <br> 

Also, should the animation be triggered on click? Also, instead of rotating the button can we only rotate the icon?

Reply:
We are already using transform to change the Y position of the button icon, so we can't overwrite it. I'll investigate another method.